### PR TITLE
GIC common code for V1V2 and V3

### PR DESCRIPTION
**Pull request by sandeepbrcm**
GIC V1 V2 V3 driver are refactored to use the common GICD code.

This reduces the code base to a good extent. There are few #if  but I feel its manageable considering the benefit of code sharing.
There are few abstractions eg: the SGI generation API and cpu id extraction. These can be useful in future if someone adds SMP
support.

### CODE REVIEW

**Comment by sandeepbrcm** [comment]
@stephanosio  This change set was dangling in my workspace during V3 implementation . Before I loose it I though to push. Please have a look if this will help. I have re-based and tested on 'qemu'.

File: include/arch/arm/aarch64/cpu.h [hunk 1]
```diff
@@ -171,6 +171,7 @@
 #define MPIDR_AFFLVL(mpidr, aff_level) \
 		(((mpidr) >> MPIDR_AFF##aff_level##_SHIFT) & MPIDR_AFFLVL_MASK)
 
-#define GET_MPIDR()		read_sysreg(mpidr_el1)
+#define get_MPIDR()		read_sysreg(mpidr_el1)
```
Comment (andyross) [inline include/arch/arm/aarch64/cpu.h:174]: Any reason for the oddball capitalization?

**Review by andyross — COMMENTED** [review]
Drive-by nitpick.  No +1 because I'm not expert enough on the architecture or GIC hardware to pontificate, but broadly it looks fine to me.

**Comment by stephanosio** [comment]
Superseded by https://github.com/zephyrproject-rtos/zephyr/pull/26681

**Pull request closed without merging**